### PR TITLE
[test](mtmv)add join infer and derive test case

### PR DESCRIPTION
**Pull request by zfr9527**
## Proposed changes

[test](mtmv)add join infer and derive test case

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zfr9527** [comment]
run buildall

**Comment by zfr9527** [comment]
run buildall

**Comment by zfr9527** [comment]
run buildall

File: regression-test/suites/nereids_rules_p0/mv/join_infer_derive/left_join_infer_and_derive.groovy [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+    def mv_stmt_4 = """select l_shipdate, o_orderdate, l_partkey, l_suppkey, orders_left.o_orderkey 
+        from lineitem_left  
+        left join orders_left 
+        on lineitem_left.l_orderkey = orders_left.o_orderkey 
+        where o_orderkey = 1"""
+    def mv_stmt_5 = """select l_shipdate, o_orderdate, l_partkey, l_suppkey, orders_left.o_orderkey 
+        from lineitem_left  
+        left join orders_left 
+        on lineitem_left.l_orderkey = orders_left.o_orderkey 
+        where l_suppkey = 2"""
```
Comment (seawinde) [inline regression-test/suites/nereids_rules_p0/mv/join_infer_derive/left_join_infer_and_derive.groovy:175]: Maybe we should have one more case like  
**mv_stmt_12**

as following  case  
select l_shipdate, o_orderdate, l_partkey, l_suppkey, orders_left.o_orderkey 
        from lineitem_left  
        left join orders_left  
        on lineitem_left.l_orderkey = orders_left.o_orderkey
        where l_suppkey = 2 and o_orderdate = '2023-10-17'

File: regression-test/suites/nereids_rules_p0/mv/join_infer_derive/right_join_infer_and_derive.groovy [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+        from lineitem_right  
+        right join orders_right 
+        on lineitem_right.l_orderkey = orders_right.o_orderkey 
+        where o_orderkey = 1"""
+    def mv_stmt_5 = """select l_shipdate, o_orderdate, l_partkey, l_suppkey, orders_right.o_orderkey 
+        from lineitem_right  
+        right join orders_right 
+        on lineitem_right.l_orderkey = orders_right.o_orderkey 
+        where l_suppkey = 2"""
+    def mv_stmt_6 = """select t.l_shipdate, o_orderdate, t.l_partkey, t.l_suppkey, orders_right.o_orderkey
```
Comment (seawinde) [inline regression-test/suites/nereids_rules_p0/mv/join_infer_derive/right_join_infer_and_derive.groovy:176]: Maybe we should have one more case like  
**mv_stmt_12**

as following  case  
```
select l_shipdate, o_orderdate, l_partkey, l_suppkey, orders_left.o_orderkey 
        from lineitem_right 
        right join orders_right  
        on lineitem_left.l_orderkey = orders_left.o_orderkey
        where l_suppkey = 2 and o_orderdate = '2023-10-17'
```

File: regression-test/suites/nereids_rules_p0/mv/join_infer_derive/left_join_infer_and_derive.groovy (hunk 1 above)
Comment (zfr9527) [reply regression-test/suites/nereids_rules_p0/mv/join_infer_derive/left_join_infer_and_derive.groovy:175]: new cases have been added

File: regression-test/suites/nereids_rules_p0/mv/join_infer_derive/right_join_infer_and_derive.groovy (hunk 2 above)
Comment (zfr9527) [reply regression-test/suites/nereids_rules_p0/mv/join_infer_derive/right_join_infer_and_derive.groovy]: new cases have been added

**Comment by zfr9527** [comment]
run buildall

**Review by seawinde — APPROVED** [review]

**Comment by zfr9527** [comment]
run buildall

**Comment by zfr9527** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Comment by morrySnow** [comment]
run p0

**Pull request merged**